src/sys/unix/net.rs

```diff
@@ -64,32 +64,76 @@ pub(crate) fn new_socket(
     socket
 }
 
-pub(crate) fn socket_addr(addr: &SocketAddr) -> (*const libc::sockaddr, libc::socklen_t) {
-    use std::mem::size_of_val;
+/// A type with the same memory layout as `libc::sockaddr`. Used in converting Rust level
+/// SocketAddr* types into their system representation. The benefit of this specific
+/// type over using `libc::sockaddr_storage` is that this type is exactly as large as it
+/// needs to be and not a lot larger. And it can be initialized cleaner from Rust.
+#[repr(C)]
+pub(crate) union SocketAddrCRepr {
+    v4: libc::sockaddr_in,
+    v6: libc::sockaddr_in6,
+}
+
+impl SocketAddrCRepr {
+    pub fn as_ptr(&self) -> *const libc::sockaddr {
```

Comment:
Nit: `pub(crate)`.

Reply:
Does that make any difference? The entire type is only crate-internal anyway, so any method on it can't possibly become public outside of the crate. I would not expect `pub(crate)` to make any difference at all here. Does it? I can change it.

Reply:
Done. I made the change. Still not sure if it has any effect, but it does not hurt, and can maybe be more readable(?)